src/argilla/training/openai.py

```diff
@@ -65,8 +72,26 @@ def __init__(self, *args, **kwargs):
             self._train_dataset = self._dataset
             self._eval_dataset = None
 
+        if not self.__legacy:
+            self._train_dataset = self._convert_data_to_new_format(self._train_dataset)
+            if self._eval_dataset is not None:
+                self._eval_dataset = self._convert_data_to_new_format(self._eval_dataset)
+
         self.init_training_args(model=self._model)
 
+    def _convert_data_to_new_format(self, dataset):
+        new_dataset = []
+        for entry in dataset:
+            new_dataset.append(
+                {
+                    "messages": [
+                        {"role": "user", "content": f"Classify the following text: {entry['prompt']}"},
+                        {"role": "assistant", "content": entry["completion"]},
+                    ]
+                }
+            )
+        return new_dataset
+
     def init_training_args(
         self,
         training_file: str = None,
```

Comment:
All the args here that have a default value of `None` which are most of them, should also update their type-hint to `Optional[...]`

```suggestion
        training_file: Optional[str] = None,
```